src/plugins/FingerPrint.js

```diff
@@ -0,0 +1,395 @@
+        headerDiv.appendChild(recognizedFPByDefaultDiv);
+        this.container.appendChild(headerDiv);
+        this.container.appendChild(document.createElement('hr'));
+
+        // body
+        const bodyDiv = document.createElement('div');
+        bodyDiv.classList.add('gm-fingerprint-dialog-body');
+
+        const buttonsDiv = document.createElement('div');
+        buttonsDiv.classList.add('gm-fingerprint-dialog-buttons');
+
+        // buttons fingerprint
+        const fingerprintButtons = [{
+            cmd: 'recognized',
+            message: FINGERPRINT_MESSAGES.toSend.SCAN_RECOGNIZED,
+            onclick: (btn) => {
+                btn.classList.add('gm-loader');
+            }
+        }, {
+            cmd: 'unrecognized',
+            message: FINGERPRINT_MESSAGES.toSend.SCAN_UNRECOGNIZED,
+        }, {
+            cmd: 'dirty',
+            message: FINGERPRINT_MESSAGES.toSend.SCAN_DIRTY,
+        }, {
+            cmd: 'too-fast',
+            message: FINGERPRINT_MESSAGES.toSend.SCAN_TOO_FAST,
+        }, {
+            cmd: 'partial',
+            message: FINGERPRINT_MESSAGES.toSend.SCAN_PARTIAL,
+        }, {
+            cmd: 'insufficient',
+            message: FINGERPRINT_MESSAGES.toSend.SCAN_INSUFFICIENT,
+        }];
+
+        fingerprintButtons.forEach((button) => {
+            this.createFingerprintButton(button, buttonsDiv);
+        });
+
+        bodyDiv.appendChild(buttonsDiv);
+        this.container.appendChild(bodyDiv);
+
+        // Footer
+        this.container.appendChild(document.createElement('hr'));
+        const footerDiv = document.createElement('div');
+        footerDiv.classList.add('gm-fingerprint-dialog-footer');
+
+        const giveFeedbackDiv = document.createElement('div');
+        giveFeedbackDiv.classList.add('gm-fingerprint-dialog-give-feedback');
+        const giveFeedbackLink = document.createElement('a');
+        giveFeedbackLink.href = 'https://github.com/orgs/Genymobile/discussions';
+        giveFeedbackLink.innerHTML = this.i18n.FINGERPRINT_GIVE_FEEDBACK_LABEL || 'Give feedback';
+        giveFeedbackLink.target = '_blank';
+        giveFeedbackLink.rel = 'noopener noreferrer';
+
+        giveFeedbackDiv.appendChild(giveFeedbackLink);
+
+        footerDiv.appendChild(giveFeedbackDiv);
+        this.container.appendChild(footerDiv);
+
+        // Setup
+        this.widget.className = 'gm-overlay gm-fingerprint-plugin gm-hidden';
+
+        // Add close button
+        const close = document.createElement('div');
+        close.className = 'gm-close-btn';
+        close.onclick = () => this.closeWidget();
+
+        this.widget.appendChild(close);
+        this.widget.appendChild(this.container);
+
+        // Render into document
+        this.overlays.push(this.widget);
+        this.instance.root.appendChild(this.widget);
+
+        // post setup
+        this.disableBody();
+    }
+
+    /**
+     * Display or hide the widget.
+     */
+    toggleWidget() {
```

Comment:
it looks like we should inheritate all this functions from a Widget parent class. We might plan this in a future code rework.

Reply:
Totally agree with that, but for me we need a dedicate story to rewrite dialog box. Somes functionnal code need to be on "OverlayPlugin" class and somes CSS too